ti/instrumentation/instrumentation.go

```diff
@@ -81,7 +88,30 @@ func getTestSelection(ctx context.Context, runner TestRunner, config *api.RunTes
 	} else {
 		log.Infoln(fmt.Sprintf("Running tests selected by Test Intelligence: %s", selection.Tests))
 	}
-	return selection
+	return selection, moduleList
+}
+
+// check if bazel optimization is enabled and call function to add new changed files to list
+func checkForBazelOptimization(ctx context.Context, workspace string, fs filesystem.FileSystem, log *logrus.Logger, files []ti.File) ([]ti.File, []string, error) {
+	var moduleList []string
+	var newFiles []ti.File
+	// check ticonfig params to allow bazel optimization, and get threshold for max file count
+	tiConfigYaml, err := getTiConfig(workspace, fs)
+	if err != nil {
+		log.Infoln("Ti config parsing before selectTests fails")
+	}
+
+	// skip bazel src inspection if optimization in config not selected
+	if tiConfigYaml.Config.BazelOptimization {
+		// add src files listed in java target rules in the BUILD.bazel files if BUILD.bazel is changed
+		newFiles, moduleList, err = addBazelFilesToChangedFiles(ctx, workspace, log, files, tiConfigYaml.Config.BazelFileCount)
```

Comment:
Can we not read integers from YAML directly? I see you are passing this as a string. The util method should take integer and callers need to make sure the input is passed correctly after validation. If validation fails, we should not panic/fail, just ignore optimization.